pkg/model/iam/iam_builder.go

```diff
@@ -244,6 +234,41 @@ func (b *IAMPolicyBuilder) BuildAWSIAMPolicy() (*IAMPolicy, error) {
 	return p, nil
 }
 
+// addEC2Permissions updates the IAM Policy with statements granting tailored
+// access to EC2 resources, depending on the instance role
+func addEC2Permissions(p *IAMPolicy, iamPrefix string, b *IAMPolicyBuilder, wildcard stringorslice.StringOrSlice) {
+	if (b.Role == api.InstanceGroupRoleNode) || (b.Role == api.InstanceGroupRoleMaster) {
+		p.Statement = append(p.Statement, &IAMStatement{
+			Effect:   IAMStatementEffectAllow,
+			Action:   stringorslice.Slice([]string{"ec2:Describe*"}),
+			Resource: wildcard,
+		})
+	}
+
+	if b.Role == api.InstanceGroupRoleMaster {
+		p.Statement = append(p.Statement,
+			&IAMStatement{
+				Effect: IAMStatementEffectAllow,
+				Action: stringorslice.Slice([]string{
+					"ec2:ModifyInstanceAttribute",
+					"ec2:CreateRoute",
+				}),
+				Resource: wildcard,
+			},
+			&IAMStatement{
+				Effect:   IAMStatementEffectAllow,
+				Action:   stringorslice.Slice([]string{"ec2:*"}),
+				Resource: wildcard,
+				Condition: Condition{
+					"StringEquals": map[string]string{
+						"ec2:ResourceTag/KubernetesCluster": b.Cluster.GetName(),
```

Comment:
This is going to force VPCs to be tagged.  I know in a perfect world the VPC should be tagged, but we are not in 
 perfect world.  I also know that we have people that do not have the VPCs tagged. Let me chew on this a bit.  I am wondering if we want to make this optional.